### DISCUSSION TITLE
PPPoE and IPoE

### DISCUSSION THREAD

**Discussion by TimOfAllTrades** [fakemanhk/openwrt-jp-ipoe#15] [General]
So the latter part of the instruction talks about using IPoE with PPPoe.  Is it really just as simple as using a switch to connect two routers to the ONU and setting one to IPoE and the other to Pppoe?  Would the IPoE settings be the same as the aforementioned instructions?

**Comment by fakemanhk** [comment]
This part was inspired by some folks using Mikrotik routers, because Mikrotik routers cannot use MAP-E, the way they use IPv6 is "bridging IPv6 WAN/LAN" + "PPPoE IPv4" and then based on IPv4/IPv6 family to route them.

Here, the OpenWrt MAP-E is a tunnel to allow users to access IPv4 with IPv6 connections, PPPoE is optional. The reason I am mentioning in last part is about service hosting, the IPv4 address from dual stack is shared among different users with different port assignment, in case you want full range PPPoE is an option. And you are right, with a switch connecting ONU you can have 2 interfaces performing IPoE & PPPoE at the same time (this is what I am doing now, I still have PPPoE with another router for home use while I am using another router to test OpenWrt IPoE)

**Comment by TimOfAllTrades** [comment]
OK I tried this last night and it seems to be working (so far).

TPlink wireless router: PPPoE set up, DHCP disabled, IPV6 disabled.
Nanopi R4s: IPoE v6 and v4 (same as your instructions), DHCP enabled

TPlink WAN -> switch to ONU
Nanopi WAN -> switch to ONU
Nanopi LAN -> TPlink LAN3

On the Nanopi I set its IP to 192.168.1.1 and the DHCP range from 192.168.1.10 to 192.168.1.100 (I probably won't have more than 90 devices at a time).
On the TPlink router, I set the IP to 192.168.1.128
The IPv4 mask for both devices is 255.255.255.0

Then I connect my PC via lan cable and wireless devices to the TPlink PPPoE router.

When I set the PC to obtain ip addresses automatically, it appears to go through the nanopi router.  I get an IPv6 address and my ipv4 is the MAP-E one when I check my IP.  Both router GUIs are accessible though 192.168.1.1 and 192.168.1.128.  For some reason, wifi devices do not get the IPv6 functionality.

If I set the PC to static IP and specify the gateway to 192.168.1.128, then it becomes connected to the PPPoE connection.  So at least now there is an option to choose which one to use by setting the gateway.

**Reply by fakemanhk** [reply]
Correct, another way to do this is, create another virtual WAN interface for PPPoE, and then with **_mwan3_** or **_pbr_** you can do more with it, and this is what I talk about at the last part of my guide.

WiFi devices, do you mean Android phones? Since ISP is using DHCPv6 which doesn't support by Android/ChromeOS devices, usual guides I found in Japanese website didn't really address this issue, and in my #2 discussion I've provided an option for SLAAC (at least my Android phones/Chromebook can get IPv6 without problem).
